src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -676,6 +679,9 @@ def _create_empty_products(self):
         or self._unused_deps_check_enabled:
       self.context.products.safe_create_data('product_deps_by_src', dict)
 
+    if self.context.products.is_required_data('zinc_args'):
```

Comment:
Can we include a TODO/ticket somewhere about extracting javac/scalac specific args out of the zinc options into their own collections on the `Java`/`Scala(Platform)` subsystems? That way you'd be able to fetch them off of the subsystem directly.

...heck... it might even be easy to warn for `-S`/`-C` prefixed options and swap them over now...?